src/snap/getAllNotifications.ts

```diff
@@ -0,0 +1,17 @@
+import { Page } from "puppeteer";
+import { clickOnElement, openProfileDropdown } from "../helpers";
+import { NotificationList } from "./types";
+
+export const getAllNotifications =
+  (page: Page) => async (): Promise<NotificationList> => {
+    await page.bringToFront();
+    await openProfileDropdown(page);
+    await clickOnElement(page, "Notifications");
+    await page.waitForTimeout(300);
```

Comment:
why the timeout? Can't we wait for selector instead?

Reply:
There is a problem here when I'm waiting for the notification-item selector, as soon as it gets the first selector it will proceed further and I will end up with only one notification in the list (the second will not render). So I'm waiting for the page to render fully and execute the search of items. The problem is that we don't know home many notifications-items we should await for.  I also don't like to use waitForTimeout, but I didn't find any solution for this case.

Reply:
How about we keep this method without awaiting, like it returns current notifications and it's up to user to figure out when they are all there?

But to help the user, we can expose the event emitter which would yield a notification event every time a notification shows up in UI. On the implementation side, if somebody registers an event listener, we would open a new tab and in it, we would open the notifications page and wait for the new element in UI.

Reply:
@mpetrunic we can do this way, but isn't it overkill? When we develop something for not stable API, it's probably better not to overcomplicate as there is a high risk of changing behavior, also I think it will not be the most popular method which brings a lot of value. Probably for InvokeSnap we could do something extra, but I'm not sure this is a case for notifications.  @BeroBurny what do you think?

Reply:
My case as snap snap developer, would you rather do:
```typescript
await dappeteer.snap.invokeSnap(...);
await sleep(300);
expect(await dappeteer.snap.getNotifications(...)).toHaveLength(2); //pray to god 300 is enough and test won't be flaky
```
or
```typescript
const emitter: EventEmitter = dappeteer.snap.notificationEmitter();
const newNotificationPromise = pEvent(emitter, "new")
await dappeteer.snap.invokeSnap(...);
const notification = await newNotificationPromise()
```

Where `notificationEmitter`should have simple implementation:
- opens new metamask tab
- opens notifications screen
- uses waitForFunction to detect change in list length: https://stackoverflow.com/questions/71685511/puppeteer-wait-for-lists-coming-up
- emit `new` event on list change

Reply:
Using the first option we need only two lines, which are pretty straightforward, we are calling the notifications and getting them from UI. The waiting time is inside the getNotifications method, we can increase it to 500 milliseconds to be on the safe side. I think it will solve 99 percent of the use cases to test notifications.

```
await dappeteer.snap.invokeSnap(...);
await dappeteer.snap.getNotifications(...)
```

There is nothing wrong with an emitter event and I totally agree that from a technical perspective it will be more "bulletproof". But we add extra work for the developer to execute one more method which is not expected in this case. If we would do our library based on emitter events, I could imagine it would be clear, but why the developer should use an emitter event specifically to get the notifications? From my experience emitters (or observable pattern) is not commonly used, so I believe it will lead to confusion rather will bring more value. 

btw, you mentioned the method pEvent(), is it from the library? https://www.npmjs.com/package/p-event

Reply:
> @mpetrunic Using the first option we need only two lines, which are pretty straightforward, we are calling the notifications and getting them from UI. The waiting time is inside the getNotifications method, we can increase it to 500 milliseconds to be on the safe side. I think it will solve 99 percent of the use cases to test notifications.
> 
> ```
> await dappeteer.snap.invokeSnap(...);
> await dappeteer.snap.getNotifications(...)
> ```
> 
> There is nothing wrong with an emitter event and I totally agree that from a technical perspective it will be more "bulletproof". But we add extra work for the developer to execute one more method which is not expected in this case. If we would do our library based on emitter events, I could imagine it would be clear, but why the developer should use an emitter event specifically to get the notifications? From my experience emitters (or observable pattern) is not commonly used, so I believe it will lead to confusion rather will bring more value.


This will also:
- slow down every test using this method for 0.5s
- there is still chance on lag inside the metamask that will cause flaky tests sometimes

But event emitter allows developer to choose. Anyways, I'm fine merging PR as is, but we should open issue to allow waiting for notification.

> 
> btw, you mentioned the method pEvent(), is it from the library? https://www.npmjs.com/package/p-event

Yup quite useful and common with async/await programing.

Reply:
MetaMask extension CI tests take 40 minutes. Any non-critical time delays are no-go.